File: CPP/Aksesoris.cpp
#include <iostream>
#include <string>
#include "Petshop.cpp"

using namespace std;

class Aksesoris : public Petshop
{

private:
    string jenis_aksesoris;
    string bahan_aksesoris;
    string warna_aksesoris;

public:
    Aksesoris()
    {
    }
    Aksesoris(string jenis_aksesoris, string bahan_aksesoris, string warna_aksesoris)
    {
        this->jenis_aksesoris = jenis_aksesoris;
        this->bahan_aksesoris = bahan_aksesoris;
        this->warna_aksesoris = warna_aksesoris;
    }
    string get_jenis_aksesoris()
    {
        return this->jenis_aksesoris;
    }
    void set_jenis_aksesoris(string jenis_aksesoris)
    {
        this->jenis_aksesoris = jenis_aksesoris;
    }
    string get_bahan_aksesoris()
    {
        return this->bahan_aksesoris;
    }
    void set_bahan_aksesoris(string bahan_aksesoris)
    {
        this->bahan_aksesoris = bahan_aksesoris;
    }
    string get_warna_aksesoris()
    {
        return this->warna_aksesoris;
    }
    void set_warna_aksesoris(string warna_aksesoris)
    {
        this->warna_aksesoris = warna_aksesoris;
    }

    ~Aksesoris()
    {
    }
};

File: PHP/Aksesoris.php
<?php

require_once 'Petshop.php';
class Aksesoris extends Petshop {
    private $jenis_aksesoris;
    private $bahan_aksesoris;
    private $warna_aksesoris;

    // Konstruktor default
    public function __construct($id = 0, $nama_produk = "", $harga_produk = 0, $stok_produk = 0, $gambar_produk = "", $jenis_aksesoris = "", $bahan_aksesoris = "", $warna_aksesoris = "") {
        parent::__construct($id, $nama_produk, $harga_produk, $stok_produk, $gambar_produk);
        $this->jenis_aksesoris = $jenis_aksesoris;
        $this->bahan_aksesoris = $bahan_aksesoris;
        $this->warna_aksesoris = $warna_aksesoris;
    }

    public function getJenisAksesoris() {
        return $this->jenis_aksesoris;
    }

    public function setJenisAksesoris($jenis_aksesoris) {
        $this->jenis_aksesoris = $jenis_aksesoris;
    }

    public function getBahanAksesoris() {
        return $this->bahan_aksesoris;
    }

    public function setBahanAksesoris($bahan_aksesoris) {
        $this->bahan_aksesoris = $bahan_aksesoris;
    }

    public function getWarnaAksesoris() {
        return $this->warna_aksesoris;
    }

    public function setWarnaAksesoris($warna_aksesoris) {
        $this->warna_aksesoris = $warna_aksesoris;
    }
}

?>
